native/cocos/bindings/manual/jsb_spine_manual.cpp

```diff
@@ -213,6 +214,314 @@ static bool js_register_spine_retainSkeletonData(se::State &s) {
+    if (nullptr == effect) return true;
+
+    float outX, outY;
+    effect->transform(outX, outY);
+
+    args[0].toObject()->setProperty("x", se::Value(outX));
+    args[0].toObject()->setProperty("y", se::Value(outY));
+    return true;
+}
+SE_BIND_FUNC(js_transform_VertexEffect)
+
+static bool js_transform_SwirlVertexEffect(se::State &s) {
+    const auto &args = s.args();
+    spine::SwirlVertexEffect* effect = SE_THIS_OBJECT<spine::SwirlVertexEffect>(s);
+    if (nullptr == effect) return true;
+
+    float outX, outY;
+    effect->transform(outX, outY);
+
+    args[0].toObject()->setProperty("x", se::Value(outX));
+    args[0].toObject()->setProperty("y", se::Value(outY));
+    return true;
+}
+SE_BIND_FUNC(js_transform_SwirlVertexEffect)
+
+static bool js_transform_JitterVertexEffect(se::State &s) {
+    const auto &args = s.args();
+    spine::JitterVertexEffect* effect = SE_THIS_OBJECT<spine::JitterVertexEffect>(s);
+    if (nullptr == effect) return true;
+
+    float outX, outY;
+    effect->transform(outX, outY);
+
+    args[0].toObject()->setProperty("x", se::Value(outX));
+    args[0].toObject()->setProperty("y", se::Value(outY));
+    return true;
+}
+SE_BIND_FUNC(js_transform_JitterVertexEffect)
+
+static bool js_spine_Skin_getAttachments(se::State& s) {
+    CC_UNUSED bool ok = true;
+    const auto& args = s.args();
+    size_t argc = args.size();
+    spine::Skin *skin = (spine::Skin *) NULL ;
+    
+    if(argc != 0) {
+        SE_REPORT_ERROR("wrong number of arguments: %d, was expecting %d", (int)argc, 0);
+        return false;
+    }
+    skin = SE_THIS_OBJECT<spine::Skin>(s);
+    if (nullptr == skin) return true;
+    spine::Skin::AttachmentMap::Entries attachments = skin->getAttachments();
+
+    std::vector<spine::Skin::AttachmentMap::Entry*> entries;
+    while (attachments.hasNext()) {
+        spine::Skin::AttachmentMap::Entry &entry = attachments.next();
+        entries.push_back(&entry);
+    }
+    
+    se::Object *array = se::Object::createArrayObject(entries.size());
+    for (int i = 0; i < entries.size(); ++i) {
+        array->setArrayElement(i, se::Value(entries[i]));
+    }
+    s.rval().setObject(array);
+    
+    return true;
+}
+SE_BIND_FUNC(js_spine_Skin_getAttachments)
+
+static bool js_spine_VertexEffect_begin(se::State& s) {
+    CC_UNUSED bool ok = true;
+    const auto& args = s.args();
+    size_t argc = args.size();
+    spine::VertexEffect *arg1 = (spine::VertexEffect *) NULL ;
+    spine::Skeleton *arg2 = 0 ;
+    
+    if(argc != 1) {
+        SE_REPORT_ERROR("wrong number of arguments: %d, was expecting %d", (int)argc, 1);
+        return false;
+    }
+    arg1 = SE_THIS_OBJECT<spine::VertexEffect>(s);
+    if (nullptr == arg1) return true;
+    
+    ok &= sevalue_to_native(args[0], &arg2, s.thisObject());
+    SE_PRECONDITION2(ok, false, "Error processing arguments");
+    
+    (arg1)->begin(*arg2);
+    
+    
+    return true;
+}
+SE_BIND_FUNC(js_spine_VertexEffect_begin) 
```

Comment:
This `begin` function could be automatically bound by using `%extend` directive in `spine.i`.